pandas/core/common.py

```diff
@@ -307,8 +312,8 @@ def _isnull_ndarraylike_old(obj):
 
     # box
     if isinstance(obj, ABCSeries):
-        from pandas import Series
-        result = Series(result, index=obj.index, name=obj.name, copy=False)
+        result = obj._constructor(
+            result, index=obj.index, name=obj.name, copy=False)
```

Comment:
these are changes which I would accept, e.g. being more generic. really really try hard not to do if/thens.